pyodide-build/pyodide_build/buildall.py

```diff
@@ -538,14 +545,25 @@ def build_from_graph(
     built_queue: Queue[BasePackage | Exception] = Queue()
     thread_lock = Lock()
     queue_idx = 1
+    building_rust_pkg = False
     progress_formatter = ReplProgressFormatter(len(needs_build))
 
     def builder(n: int) -> None:
-        nonlocal queue_idx
+        nonlocal queue_idx, building_rust_pkg
         while True:
-            pkg = build_queue.get()[1]
+            _, pkg = build_queue.get()
 
             with thread_lock:
+                if is_rust_package(pkg):
+                    # Don't build multiple rust packages at the same time
+                    # See: https://github.com/pyodide/pyodide/issues/3565
+                    if building_rust_pkg:
+                        build_queue.put((job_priority(pkg), pkg))
+                        sleep(0.01)
```

Comment:
What's this sleep do? It needs a comment I think.

Reply:
It releases the GIL so that main thread can queue new packages, following the same step done in [here](https://github.com/pyodide/pyodide/blob/205deef3ccc38b32b510e2854bf12f73e9b332c8/pyodide-build/pyodide_build/buildall.py#L572-L573), will add a comment.